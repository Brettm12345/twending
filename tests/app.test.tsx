/* eslint-disable no-unused-vars, @typescript-eslint/no-unused-vars */
import React from 'react'
import { create } from 'react-test-renderer'
import Home from '../src/pages'

it('Renders without an error', () => {
  expect(create(<Home />).toJSON()).toMatchSnapshot()
})
