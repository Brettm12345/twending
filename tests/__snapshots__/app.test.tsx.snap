// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without an error 1`] = `
<main
  className="justify-center items-center flex flex-col pt-24 mb-10"
>
  <header
    className="fixed top-0 right-0 z-50 flex w-full bg-gray-800 shadow-md"
  >
    <div
      className="relative flex justify-between items-center w-full h-20 px-4"
    >
      <a
        className="mr-auto opacity-75 transition-opacity hidden sm:flex"
        href="https://github.com/brettm12345/twending"
      >
        <svg
          fill="none"
          height="3em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={1}
          viewBox="0 0 24 24"
          width="3em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
      </a>
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        className="sc-fznZeY kkGVfQ rfs-select-container"
        data-testid="rfs-select-container"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="sc-fzqBZW llpyUy rfs-control-container"
          data-testid="rfs-control-container"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="sc-fzokOt fwJBYx"
          >
            <div
              className="sc-AxhUy haooFA"
            >
              <span
                className="inline-flex mr-4 ellipsis text-gray-100"
              >
                <span
                  className="language__dot-sc-1m7m30k-0 bIRSVm inline-block mr-1 border relative rounded-full border-gray-700"
                  style={
                    Object {
                      "backgroundColor": "var(--gray-400)",
                    }
                  }
                />
                All Languages
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className="sc-Axmtr cOUxKw rfs-autosize-input"
                data-testid="rfs-autosize-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={false}
                spellCheck={false}
                style={
                  Object {
                    "width": 2,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="sc-AxheI dyhkWB"
              >
                
              </div>
            </div>
          </div>
          <div
            className="sc-fzozJi kDA-dBh"
          >
            <div
              className="sc-fznyAO dPBQkx"
            />
            <div
              aria-hidden="true"
              className="sc-fzoLsD fdZUZv"
              data-testid="rfs-caret-icon"
            >
              <svg
                aria-hidden={true}
                className="fill-current stroke-current inline-block leading-none transition-transform"
                focusable={false}
                height={20}
                strokeWidth={0}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr esrIbz rfs-menu-container"
          data-testid="rfs-menu-container"
          onMouseDown={[Function]}
        >
          <div
            onScroll={[Function]}
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "direction": "ltr",
                "height": 0,
                "overflow": "auto",
                "position": "relative",
                "width": "100%",
                "willChange": "transform",
              }
            }
          >
            <div
              style={
                Object {
                  "height": 0,
                  "pointerEvents": undefined,
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="sc-AxjAm iiEGxM"
          >
            No options
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        className="sc-fznZeY kkGVfQ rfs-select-container"
        data-testid="rfs-select-container"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="sc-fzqBZW llpyUy rfs-control-container"
          data-testid="rfs-control-container"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="sc-fzokOt fwJBYx"
          >
            <div
              className="sc-AxhUy haooFA"
            >
              Monthly
            </div>
            <div
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className="sc-Axmtr cOUxKw rfs-autosize-input"
                data-testid="rfs-autosize-input"
                onBlur={[Function]}
                onFocus={[Function]}
                readOnly={true}
                spellCheck={false}
                style={
                  Object {
                    "width": 2,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="sc-AxheI dyhkWB"
              >
                
              </div>
            </div>
          </div>
          <div
            className="sc-fzozJi kDA-dBh"
          >
            <div
              className="sc-fznyAO dPBQkx"
            />
            <div
              aria-hidden="true"
              className="sc-fzoLsD fdZUZv"
              data-testid="rfs-caret-icon"
            >
              <svg
                aria-hidden={true}
                className="fill-current stroke-current inline-block leading-none transition-transform"
                focusable={false}
                height={20}
                strokeWidth={0}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr esrIbz rfs-menu-container"
          data-testid="rfs-menu-container"
          onMouseDown={[Function]}
        >
          <div
            onScroll={[Function]}
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "direction": "ltr",
                "height": 0,
                "overflow": "auto",
                "position": "relative",
                "width": "100%",
                "willChange": "transform",
              }
            }
          >
            <div
              style={
                Object {
                  "height": 0,
                  "pointerEvents": undefined,
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="sc-AxjAm iiEGxM"
          >
            No options
          </div>
        </div>
      </div>
    </div>
  </header>
  <h1
    className="mt-6 text-2xl text-center"
  >
    Trending Repositories
  </h1>
  <svg
    aria-label="Loading"
    height={80}
    stroke="var(--blue)"
    viewBox="0 0 45 45"
    width={80}
  >
    <g
      fill="none"
      fillRule="evenodd"
      strokeWidth={2}
      transform="translate(1 1)"
    >
      <circle
        cx={22}
        cy={22}
        r={6}
        strokeOpacity={0}
      >
        <animate
          attributeName="r"
          begin="1.5s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="6;12"
        />
        <animate
          attributeName="stoke-opacity"
          begin="1.5s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="1;0"
        />
        <animate
          attributeName="stroke-width"
          begin="1.5s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="2;0"
        />
      </circle>
      <circle
        cx={22}
        cy={22}
        r={6}
        strokeOpacity={0}
      >
        <animate
          attributeName="r"
          begin="3s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="6;12"
        />
        <animate
          attributeName="stoke-opacity"
          begin="3s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="1;0"
        />
        <animate
          attributeName="stroke-width"
          begin="3s"
          calcMode="linear"
          dur="3s"
          repeatCount="indefinite"
          values="2;0"
        />
      </circle>
      <circle
        cx={22}
        cy={22}
        r={8}
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="1.5s"
          repeatCount="indefinite"
          values="6;1;2;3;4;5;6"
        />
      </circle>
    </g>
  </svg>
  <div
    className="mt-6"
  >
    <button
      className="px-4 py-2 font-semibold rounded text-white bg-blue duration-300 ease-in-out transition-all hover:bg-lightBlue transform hover:shadow-xl hover:-translate-y-1 focus:outline-none"
      onClick={[Function]}
    >
      Load next month
    </button>
  </div>
</main>
`;
